### PR TITLE
feat: search filters from oak components - LESQ-766

### PR DESCRIPTION
**Pull request by JBR90**
## Description

Music year: {owa_music_year}

- uses Oak Components search filter checkbox
- Updated designs for search page 
- Move Unit and Lesson below search bar
- Refactor search events for new component 

## Issue(s)

Fixes #

## How to test

1. Go to https://deploy-preview-2400--oak-web-application.netlify.thenational.academy
Go to search page - checkbox's should be new Oak Search Checkbox
Events searchRefined should be sent to avo / posthog

## Checklist

- [ ] Added or updated tests where appropriate
- [ ] Manually tested across browsers / devices
- [ ] Considered impact on accessibility
- [ ] Design sign-off
- [ ] Approved by product owner
- [ ] Does this PR update a package with a breaking change

### CODE REVIEW

File: src/components/TeacherComponents/SearchFilters/SearchFilters.tsx [hunk 1]
```diff
@@ -20,88 +29,115 @@ const SearchFilters: FC<SearchFiltersProps> = (props) => {
...
-          Key stage
-        </OakP>
-        <OakFlex
+    <OakThemeProvider theme={oakDefaultTheme}>
+      <OakBox>
+        <OakBox
+          $mb="space-between-m2"
+          $bb={"border-solid-s"}
+          $borderColor={"grey40"}
+          // $mt={["space-between-l", null]}
```
Comment (k-huggs) [inline src/components/TeacherComponents/SearchFilters/SearchFilters.tsx:44]: could remove comment if not required

**Review by Monika-Koziol — CHANGES_REQUESTED** [review]
Filters are looking good. There were a couple of weird things that might not be part of this ticket:

- [ ] strange cropped 'no results' underneath the nav, see attached screenshot
Screenshot 2024-04-30 at 16 19 25

- [ ] is it possible to have [showing X results] closer to the unit/lesson filter with a larger gap away from the actual results? I can write another ticket for that if this was not part of this one.

**Comment by JBR90** [comment]
@Monika-Koziol - [ ] is it possible to have [showing X results] closer to the unit/lesson filter with a larger gap away from the actual results? I can write another ticket for that if this was not part of this one.

Sure, what distance do you want it be be ?

**Review by k-huggs — APPROVED** [review]

**Review by jack-skerrett-oak — CHANGES_REQUESTED** [review]
There are some general spacing issues with the top part of search that could potentially do with some looking at, but I might post about that in the squad channel, I know Monika mentioned it.

- [ ] I think we could get rid of the alt text on the subject icons on the search filters

**Comment by JBR90** [comment]
@jack-skerrett-oak Is Alt="" ok?

**Comment by jack-skerrett-oak** [comment]
@JBR90 Yep should be!

**Comment by JBR90** [comment]
> @jack-skerrett-oak Is Alt="" ok?

Will happen with the next component release

**Review by jack-skerrett-oak — APPROVED** [review]
With the understanding that the alt text will be changed at next release due to oak-components NPM weirdness and that we will be defining better spacing on another ticket because it isn't all that relevant to this ticket.

**Review by Monika-Koziol — APPROVED** [review]

**Pull request merged**